src/RevProc.cc

```diff
@@ -1337,31 +1330,34 @@ RevInst RevProc::DecodeInst(){
   }
 
   // Stage 4: Determine if we have a funct7 field (R-Type and some specific I-Type)
-  uint32_t Funct7 = 0x00ul;
+  uint32_t Funct2or7 = 0x00ul;
   if( inst65 == 0b01 ) {
     if( (inst42 == 0b011) || (inst42 == 0b100) || (inst42 == 0b110) ){
       // R-Type encodings
-      Funct7 = ((Inst >> 25) & 0b1111111);
+      Funct2or7 = ((Inst >> 25) & 0b1111111);
       //Atomics have a smaller funct7 field - trim out the aq and rl fields
       if(Opcode == 0b0101111){
-        Funct7 = (Funct7 &0b01111100) >> 2;
+        Funct2or7 = (Funct2or7 &0b01111100) >> 2;
       }
     }
+  }else if((inst65== 0b10) && (inst42 < 0b100)){
+    // R4-Type encodings -- we store the Funct2 precision field in Funct2or7
+    Funct2or7 = DECODE_FUNCT2(Inst);
```

Comment:
This is where `Funct2or7` is set to `Funct2` (the precision), when `Inst 4:2` is either `0b00`, `0b01`, `0b10`, or `0b11` (less than `0b100`), corresponding to the 4 REV-4 instructions `fmadd`, `fmsub`, `fnmadd`, `fnmsub`.